experiment/simulation/js/animation.js

```diff
@@ -0,0 +1,494 @@
+import { setCoordinates, fillInputDots, objectDisappear, objectAppear, fillColor, setColor, unsetColor, getXor } from "./animation-utility.js";
+'use strict';
+
+window.simulationStatus = simulationStatus;
+window.restartCircuit = restartCircuit;
+window.setSpeed = setSpeed;
+// Dimensions of working area
+const circuitBoard = document.getElementById("circuit-board");
+const sidePanels = document.getElementsByClassName("v-datalist-container");
+
+// Distance of working area from top
+const circuitBoardTop = circuitBoard.offsetTop;
+
+// Full height of window
+const windowHeight = window.innerHeight;
+const width = window.innerWidth;
+const svg = document.querySelector(".svg");
+const svgns = "http://www.w3.org/2000/svg";
+
+const EMPTY = "";
+const status = document.getElementById("play-or-pause");
+const observ = document.getElementById("observations");
+const speed = document.getElementById("speed");
+
+let currPos = 0;
+
+const OBJECTS = [
+    document.getElementById("x"), 
+    document.getElementById("y"), 
+    document.getElementById("clock"), 
+    document.getElementById("a")
+];
+const ARRAYX = [
+    document.getElementById("x1"), 
+    document.getElementById("x2"), 
+    document.getElementById("x3"), 
+    document.getElementById("x4"), 
+    document.getElementById("x5"), 
+    document.getElementById("x6"), 
+    document.getElementById("x7"), 
+    document.getElementById("x8"), 
+    document.getElementById("x9"), 
+    document.getElementById("x10"), 
+    document.getElementById("x11"), 
+    document.getElementById("x12")
+];
+const ARRAYY = [
+    document.getElementById("y1"),
+    document.getElementById("y2"), 
+    document.getElementById("y3"), 
+    document.getElementById("y4"), 
+    document.getElementById("y5"), 
+    document.getElementById("y6"), 
+    document.getElementById("y7"), 
+    document.getElementById("y8"), 
+    document.getElementById("y9"), 
+    document.getElementById("y10"), 
+    document.getElementById("y11"), 
+    document.getElementById("y12")
+];
+const TEXTINPUT = [
+    document.createElementNS(svgns, "text"), 
+    document.createElementNS(svgns, "text")
+];
+const TEXTCLOCK = [document.createElementNS(svgns, "text")];
+const TEXTOUTPUT = [document.createElementNS(svgns, "text")];
+const INPUTDOTS = [
```

Comment:
Multiple variables with all capitals, please make sure that these are constants before using the all CAPITALS. It is correct in the case of EMPTY, other variables I am not very sure about.

Reply:
changed to camel Case